lib/vm/src/vmcontext.rs

```diff
@@ -421,14 +422,42 @@ mod test_vmtable_definition {
     }
 }
 
+/// A typesafe wrapper around the storage for a global variables.
+///
+/// # Safety
+///
+/// Accessing the different members of this union is always safe because there
+/// are no invalid values for any of the types and the whole object is
+/// initialized by VMGlobalDefinition::new().
+#[derive(Clone, Copy)]
+#[repr(C, align(16))]
+pub union VMGlobalDefinitionStorage {
+    as_i32: i32,
+    as_u32: u32,
+    as_f32: f32,
+    as_i64: i64,
+    as_u64: u64,
+    as_f64: f64,
+    as_u128: u128,
```

Comment:
I'm very confident this not what we want.
This will duplicate the data in the structure.

If you change bytes, all the representations shall change.

Reply:
That means the `as_i32`, `as_...` should just be views over the provided storage

Reply:
This is a union, not a struct. The in-memory representation hasn't changed. The generated code from the compilers still works fine.

Reply:
What kind of magic trick is this one?

Reply:
A `union` is `mem::transmute` in disguise. The union has the size of the largest of its members (not the sum) and the alignment of the max of its members. So all of these are sharing the same [u8; 16] space (though some of them like i32 are smaller and only using part of the space, but that's fine). As long as it's safe to transmute -- which it is because these are all primitive data types with no possible invalid values and because we initialized the memory -- it's safe read or write through any of these at any time.